app/models/application.rb

```diff
@@ -31,6 +31,11 @@ def source_must_be_compatible
   def superkey_workflow
     return unless source.app_creation_workflow == SUPERKEY_WORKFLOW
 
+    if source.super_key.nil?
+      update!(:availability_status => "unavailable", :availability_status_error => "Superkey Credential Missing")
```

Comment:
Because this error can be shown in the UI, maybe we could make it more user friendly and tell users what to do.

example:
"The source is missing credentials for account authorization. Please remove the source and try to add it again / open a ticket to solve this issue."

Reply:
You got it! Updating.